src/CloudEvent/V1/Event/Data.hs

```diff
@@ -0,0 +1,200 @@
+  ( AllUniqueLabels r
+  , Forall r FromJSON
+  , Forall r ToJSON
+  )
+  => Binary (CloudEventRec r)
+
+data MkCloudEventArg = MkCloudEventArg
+  { id :: Text
+  -- ^ Unique identifier for the event that enables event traceability and deduplication
+  , source :: Text -- TODO: Replace with Iri
+
+  -- ^ Identifies where the event originated, such as a system, service, or component. Must be an absolute URI
+  , eventType :: Text
+  -- ^ Describes what kind of event occurred, typically using a domain-specific convention. Event versioning info can be included here [More Info](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/primer.md#the-role-of-the-type-attribute-within-versioning)
+  , dataContentType :: Maybe Text
+  -- ^ Format of the event data (e.g., "application/json") to help receivers parse it correctly. MIME type for the event data
+  , dataSchema :: Maybe Iri
+  -- ^ Reference to a schema that describes the structure and validation rules for the event data. Must be an absolute URI
+  , subject :: Maybe Text
+  -- ^ Specific subject or target that the event relates to within the event source's context
+  , time :: Text -- TODO replace with UTCTime
+
+  -- ^ When the event occurred, helping establish chronology in event sequences
+  }
+
+mkCloudEventRec
+  :: ( Rec.FromNative extensionAttributesNativeRecord
+     , Rec.NativeRow extensionAttributesNativeRecord ~ extensionAttributesRows
+     )
+  => MkCloudEventArg
+  -> eventData
+  -> extensionAttributesNativeRecord
+  -> CloudEventRec (CloudEventRows eventData .+ extensionAttributesRows)
+mkCloudEventRec x eventData extensionAttributesNativeRecord =
+  let extensionAttributesRec = Rec.fromNative extensionAttributesNativeRecord
+  in  CloudEventRec
+        $ #ceId .== x.id
+          .+ #ceSource .== x.source
+          .+ #ceSpecVersion .== ("1.0" :: Text)
+          .+ #ceType .== x.eventType
+          .+ #ceDataContentType .== x.dataContentType
+          .+ #ceDataSchema .== x.dataSchema
+          .+ #ceSubject .== x.subject
+          .+ #ceTime .== x.time
+          .+ #ceData .== eventData
+          .+ #ceDataBase64 .== (Nothing :: Maybe Text)
+          .+ extensionAttributesRec
+
+getCloudEventRec :: CloudEventRec r -> Rec r
+getCloudEventRec = (.getRec)
+
+ceQuery :: KnownSymbol l => CloudEventRec r -> Label l -> r .! l
+ceQuery r l = getCloudEventRec r .! l
+
+ceUpdate
+  :: (KnownSymbol l, r .! l ≈ a)
+  => Label l -> a -> CloudEventRec r -> CloudEventRec r
+ceUpdate l v r =
+  let
+    rec = getCloudEventRec r
+    updatedRec = Rec.update l v rec
+  in
+    CloudEventRec updatedRec
+
+ceFocus
+  :: ( Functor f
+     , KnownSymbol l
+     , r .! l ≈ a
+     , r ~ Rec.Modify l a r'
+     , r' .! l ≈ b
+     , r' ~ Rec.Modify l b r
+     )
+  => Label l -> (a -> f b) -> CloudEventRec r -> f (CloudEventRec r')
+ceFocus l f r =
+  let
+    rec = getCloudEventRec r
+    updatedRec = Rec.focus l f rec
+  in
+    CloudEventRec <$> updatedRec
```

Comment:
I haven't the slightest idea what's going on here. Does this function do something complicated?

Reply:
With great power of types comes great responsibility to write documentation and usage examples

Reply:
It's kinda like `zoom` from the curricula project

Reply:
Yeah, I should add more documentation. I kinda wanted to get the team's thoughts on the overall approach first